var/spack/repos/builtin/packages/py-aiofiles/package.py

```diff
@@ -15,11 +15,15 @@ class PyAiofiles(PythonPackage):
 
     license("Apache-2.0")
 
-    version("0.7.0", sha256="a1c4fc9b2ff81568c83e21392a82f344ea9d23da906e4f6a52662764545e19d4")
-    version("0.5.0", sha256="98e6bcfd1b50f97db4980e182ddd509b7cc35909e903a8fe50d8849e02d815af")
-
-    depends_on("python@3.6:3", when="@0.7:", type=("build", "run"))
```

Comment:
Seems like we lost the upper bound on Python 3

Reply:
I'm adding constraints like this: collect a list of versions for which the constraint holds, then simplify to ranges. In this case the marker has `python_version >= "3.6" and python_version < "4"`, which holds for all python versions 3.6.0, ..., 3.13.0 (or whatever the latest is). It gets simplified to a range `@3.6:` w/o upperbound since that's unnecessary.

Reply:
Can we keep the upperbound and simplify `<4` to `<=3`?

Reply:
could add a fake python 4.0.0 version, but I don't really see the point to be honest.

Reply:
It's only for theoretical future backwards-compatibility. If everything is auto-generated, it isn't hard to update it in the event that Python 4 is ever released...

Reply:
I've added a fake Python 4.0.0 so `python@:3` type bounds are added.

Reply:
That seems like a significantly worse workaround for an easy to solve issue...